src/System.Private.Xml/tests/XmlReader/Tests/ReaderEncodingTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Xunit;
+
+namespace System.Xml.Tests
+{
+    /// <summary>
+    /// This class is not completely testing XmlReader Encoding, it has a regression tests for the fix of the issue: https://github.com/dotnet/corefx/issues/35073
+    /// which reported due to fuzzy testing. Defect happening while encoding byte array, which includes a surrogate char and an invalid char.
+    /// </summary>
+    public class ReaderEncodingTests
+    {
+        private static string _invalidCharMessageStart = "Data at the root level is invalid";
+        private static string _badStartNameChar = "Name cannot begin with the";
+        private static string _invalidCharInThisEncoding = "Invalid character in the given encoding";
+
+        [Fact]
+        public static void ReadWithSurrogateCharAndInvalidChar()
+        {
+            // {60, 0, 0, 0} is a normal char, {0, 34, 1, 0}  is a surrogate char {62, 100, 60, 47} is an invalid char
+            var bytes = new byte[] { 60, 0, 0, 0, 0, 34, 1, 0, 62, 100, 60, 47, 97, 62, 10 };
+            var reader = XmlReader.Create(new MemoryStream(bytes));
+
+            XmlException ex = Assert.Throws<XmlException>(() => reader.Read());
+            Assert.Contains(_invalidCharInThisEncoding, ex.Message);
+            Assert.Equal(4, ex.LinePosition);
+        }
+
+        [Fact]
+        public static void ReadWithNormalCharAndInvalidChar()
+        {
+            // {60, 0, 0, 0, 65, 0, 0, 0} are normal chars, {62, 100, 60, 47} is an invalid char, similar bytes used below tests
+            var bytes = new byte[] { 60, 0, 0, 0, 65, 0, 0, 0, 62, 100, 60, 47, 97, 62, 10 };
+            var reader = XmlReader.Create(new MemoryStream(bytes));
+
+            XmlException ex = Assert.Throws<XmlException>(() => reader.Read());
+            Assert.Contains(_invalidCharInThisEncoding, ex.Message);
+            Assert.Equal(3, ex.LinePosition);
+        }
+
+        [Fact]
+        public static void ReadWithSurrogateCharAndInvalidChar_ValidXmlStructure()
```

Comment:
`ReadWithSurrogateCharAndInvalidChar_ValidXmlStructure` and `ReadWithSurrogateCharAndInvalidChar` failing for netfx with `System.IndexOutOfRangeException`:

 (https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F36159~2Fmerge/test~2Ffunctional~2Fcli~2F/20190319.19/workItem/System.Xml.RW.XmlReader.Tests)



```
   at System.Xml.Ucs4Decoder.Ucs4ToUTF16(UInt32 code, Char[] chars, Int32 charIndex)
   at System.Xml.Ucs4Decoder4321.GetFullChars(Byte[] bytes, Int32 byteIndex, Int32 byteCount, Char[] chars, Int32 charIndex)
   at System.Xml.Ucs4Decoder.Convert(Byte[] bytes, Int32 byteIndex, Int32 byteCount, Char[] chars, Int32 charIndex, Int32 charCount, Boolean flush, Int32& bytesUsed, Int32& charsUsed, Boolean& completed)
   at System.Xml.XmlTextReaderImpl.InvalidCharRecovery(Int32& bytesCount, Int32& charsCount)
   at System.Xml.XmlTextReaderImpl.GetChars(Int32 maxCharsCount)
   at System.Xml.XmlTextReaderImpl.ReadData()
   at System.Xml.XmlTextReaderImpl.FinishInitStream()
   at System.Xml.XmlReaderSettings.CreateReader(Stream input, Uri baseUri, String baseUriString, XmlParserContext inputContext)
   at System.Xml.Tests.ReaderEncodingTests.ReadWithSurrogateCharAndInvalidChar_ValidXmlStructure() in D:\a\1\s\src\System.Private.Xml\tests\XmlReader\Tests\ReaderEncodingTests.cs:line 48
```

Reply:
yeah need to skip framework tests